services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_insights/history.tsx

```diff
@@ -0,0 +1,201 @@
+const accountImg = <img alt="" src={`${EXTRA_SMALL_ICON_BASE_SRC}/account-green.svg`} />
+
+interface HistoryProps {
+  classroom: ClassroomInterface;
+  useELLDiagnostics: boolean;
+  timeframe: string;
+}
+
+const headers = [
+  {
+    name: 'Student',
+    width: '350px',
+    attribute: 'student',
+    isSortable: true,
+    sortAttribute: 'nameForSort'
+  },
+  {
+    name: 'Starter Diagnostic',
+    width: '270px',
+    attribute: 'starter',
+    noTooltip: true
+  },
+  {
+    name: 'Intermediate Diagnostic',
+    width: '270px',
+    attribute: 'intermediate',
+    noTooltip: true
+  },
+  {
+    name: 'Advanced Diagnostic',
+    width: '270px',
+    attribute: 'advanced',
+    noTooltip: true
+  },
+]
+
+const History = ({ classroom, timeframe, useELLDiagnostics, }: HistoryProps) => {
+  const [loading, setLoading] = React.useState<boolean>(true);
+  const [studentHistories, setStudentHistories] = React.useState<StudentHistoryInterface[]>(null)
+
+  React.useEffect(() => {
+    setLoading(true)
+    const queryHash = { timeframe }
+    if (useELLDiagnostics) {
+      queryHash['ell'] = true
+    }
+    const url = queryString.stringifyUrl({ url: `/diagnostic_insights/history_by_classroom/${classroom.id}`, query: queryHash }, { arrayFormat: 'bracket' })
+
+    requestGet(url, (body) => {
+      setStudentHistories(body)
+      setLoading(false)
+    })
+  }, [classroom, timeframe, useELLDiagnostics])
+
+  if (loading) {
+    return (
+      <div className="diagnostic-insights-content history by-class">
+        <Spinner />
+      </div>
+    )
+  }
+
+
+  const renderDiagnosticSquares = (studentLearningSequences: StudentLearningSequenceInterface[]) => {
+    if (!studentLearningSequences) { return null }
+
+    return studentLearningSequences.map(studentLearningSequence => {
+      const { student_learning_sequence_activities, } = studentLearningSequence
+      const { pre_diagnostic, post_diagnostic, pre_recommendation, post_recommendation, } = student_learning_sequence_activities
+
+      // sometimes there is a studentLearningSequence when a student has completed an activity that belongs to the recommendation sequence,
+      // even if they didn't complete either of the diagnostics, in which case we don't want to render anything
+      if (!pre_diagnostic) { return null }
+
+      let icon: JSX.Element
+
+      if (pre_diagnostic && post_diagnostic) {
+        icon = pre_diagnostic[0].proficient || post_diagnostic[0].proficient ? filledBlueDiamondWithCheckMarkImg : filledBlueDiamondImg
+      } else {
+        icon = pre_diagnostic[0].proficient ? outlinedBlueDiamondWithCheckMarkImg : outlinedBlueDiamondImg
+      }
+
+      const tooltipText = []
+
+      // Baseline Pre Completed
+      if (pre_diagnostic?.length) {
+        const preDiagnosticCompletionDate = moment(pre_diagnostic[pre_diagnostic.length - 1].completed_at).format("MM/DD/YY");
```

Comment:
We could put this format param in a constant somewhere, since it's being used many places